group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2093,6 +2095,25 @@ private List<StreamsGroupHeartbeatResponseData.TaskIds> createStreamsGroupHeartb
             .collect(Collectors.toList());
     }
 
+    private List<StreamsGroupHeartbeatResponseData.EndpointToPartitions> maybeBuildEndpointToPartitions(StreamsGroup group) {
```

Comment:
We definitely want to avoid building this map in every heartbeat. I suppose this is anyways going to be fixed with the new epoch?

Reply:
Yes absolutely just for this PR - next up is only sending when the assignment changes